Stop persisting supporting page admin data

bensymonds opened the pull request:
I had a hard time giving this a title, but basically we send supporting data to/from the client _inside_ the page json, which means it ends up getting persisted in the database. This means (a) we store a load of stuff we don't need to and (b) we can't turn off Mongoid dynamic fields (because obviously there are no fields defined for this data).

Currently this branch just deletes on the way back the data that was inserted on the way out. Pretty hacky. I'm not really convinced it's good enough to go in, but maybe it's an ok stopgap. There are better solutions, but needs more discussion, so this PR can be the start of it.

bensymonds committed:
Don't persist `type` in `Slice` records in db

`type` is inserted into json responses for the admin ui. It is generated dynamically so shouldn't be stored.

bensymonds committed:
Don't persist `client_id` in `Slice` records in db

erino commented:
:+1:

bensymonds commented:
@erino you mean you think we should just merge it?

bensymonds committed:
slice name/template from client

bensymonds committed:
`_deleted` was replaced by `_destroy` ages ago

bensymonds committed:
Set `allow_dynamic_fields` to false in Mongoid config

bensymonds commented:
I think we should add v2.0 milestone to this. It's pretty crappy to be accidentally persisting this stuff.

bensymonds commented:
Also it's been soaked in production for a good while now.

erino commented on spec/models/slice/as_json_spec.rb:
```diff
@@ -4,8 +4,7 @@
   describe "#as_json" do
 
     let :slice do
-      Slice.new(title: 'Title',
```
It would be useful to keep a spec to ensure that attributes defined in [slices](https://github.com/withassociates/slices/blob/master/spec/dummy/app/slices/title/title_slice.rb#L2) are included when the slice is serialized.

bensymonds replied:
Ok yeah, this is just in a `wip` commit so not sure what I was thinking. I guess `title` is not a field in `Slice` so it was causing the dynamic attr error, so I just removed it. But actually what we want to do is keep `title` and change `Slice` to `TitleSlice`. :+1:

bensymonds commented:
So my thoughts from the PR description at the top still stand - this is a pretty crappy way of fixing the issue. It means that any time a Slices user adds their own stuff to the page json (as we do in a lot of projects), they have to add the equivalent code to remove it when saving. @thomasmarshall I think you have been doing this in some of your projects - how do you feel about it?

We probably need to have a conversation in real life about other, better solutions. But is this approach good enough to go into `master` for now??

bensymonds committed:
Use a slice class that has the correct attributes

bensymonds committed:
Fix messed up slice/attachment attributes

Most of them are no longer present in the slice/attachment. Some are in the wrong place.

bensymonds committed:
Don't persist supporting data for admin ui in `Page`s in db

bensymonds commented:
For what it's worth, I've reworded and cleaned up the commits on this, so if we do decide to go with this approach, it's good to go (to review, I mean - especially the `client_id` bit).

bensymonds commented:
I need to pick this up again.

bensymonds commented:
I am not going to pick this up again.
